backend/organization/utility/requests.py

```diff
@@ -0,0 +1,156 @@
+                                          ,target_project_id=self.project
+                                          ,target_organization_id=self.organization).count()
+
+
+            
+            if n > 0:
+                self.validation_failed = True 
+                self.errors.append(Exception("Request Already Exists"))
+                self.duplicate_request = True
+
+
+
+
+
+        else: #id of request is supplied
+            self.membership_request = MembershipRequests.objects.filter(id=int(kwargs['membership_request_id']))
+            if self.membership_request.count()!=1 :
+                self.corrupt_membership_request_id = True
+                self.validation_failed = True 
+                self.errors.append(Exception(f"More than a record or not a single record was found in membership requests for request id {int(kwargs['membership_request_id'])}"))
+            else:
+                self.membership_request = self.membership_request.first()
+                self.corrupt_membership_request_id = False
+                self.membership_target = MembershipTarget(self.membership_request.target_membership_type)
+                self.user = self.membership_request.user
+                self.user_availability = self.membership_request.availability
+                self.project = self.membership_request.target_project
+                if 'project' in kwargs:
+                    if kwargs['project'].id != self.membership_request.target_project.id: 
+                        self.validation_failed = True
+                        self.errors.append(Exception("Inconsistent Project and Request"))
+
+        return
+
+
+
+    def create_membership_request(self,**kwargs):
+        if self.duplicate_request: return False 
+        request_obj = MembershipRequests.objects.create(user=self.user
+                                                        ,target_membership_type=self.membership_target.value
+                                                        ,target_project=self.project 
+                                                        ,target_organization=self.organization
+                                                        ,availability=self.user_availability
+                                                        ,requested_at=timezone.now()
+                                                        ,message=self.message
+                                                        ,request_status=RequestStatus.PENDING.value                                                    
+                )
+        request_id = request_obj.id
+
+
+
+        return request_id
+
+
+
+
+    def approve_request(self,**kwargs):
+
+
+        self.membership_request.request_status = RequestStatus.APPROVED.value 
+        self.membership_request.approved_at = timezone.now()
+
+        #create a record in ProjectMember
+        user_role = Role.objects.filter(name="Member").first() 
```

Comment:
Here I'd also filter by role_type=Role.READ_ONLY_TYPE instead of filtering by name.